syntax = "proto3";

package ou_pokemon;

service OUPokemanGame
{
    rpc Board(Hostname) returns (InitialMoves){}
    rpc TrainerMove(TrainMove) returns (Message){}
    rpc PokemonMove(PokMove) returns (Message){}
}

message Hostname {
    string name = 1;
}

message InitialMoves {
    string exist = 1;
}

message PokMove {
    string hostname = 1;
}

message TrainMove {
    string hostname = 1;
}

message Message {
    int32 count = 1;
}
